shared/src/types/vote_extensions/validator_set_update.rs

```diff
@@ -301,6 +301,38 @@ fn compute_hash(
     ])
 }
 
+/// Struct for serializing validator set
+/// arguments with ABI for Ethereum smart
+/// contracts.
+#[derive(Debug, Clone, Default)]
+pub struct ValidatorSetArgs {
+    /// Ethereum address of validators
+    pub validators: Vec<EthAddress>,
+    /// Voting powers of validators
+    pub powers: Vec<Uint>,
+    /// A nonce
+    pub nonce: Uint,
+}
+
+impl Encode<1> for ValidatorSetArgs {
+    fn tokenize(&self) -> [Token; 1] {
+        let addrs = Token::Array(
+            self.validators
+                .iter()
+                .map(|addr| Token::Address(addr.0.into()))
+                .collect(),
+        );
+        let powers = Token::Array(
+            self.powers
+                .iter()
+                .map(|power| Token::Uint(power.clone().into()))
+                .collect(),
+        );
+        let nonce = Token::Uint(self.nonce.clone().into());
+        [Token::FixedArray(vec![addrs, powers, nonce])]
```

Comment:
```suggestion
        [Token::Tuple(vec![addrs, powers, nonce])]
```
I think it should be a tuple. a `FixedArray` would be wrong, because it would be receiving heterogeneous type args.

Reply:
I'm also not sure if this serializes to the same type anyway, since it's not a solidity primitive...

Reply:
I have no idea. I asked @Fraccaman  and this is what he said. Probably won't know for sure until we do integrations.